src/components/chartRepo/ChartRepo.tsx

```diff
@@ -3,28 +3,59 @@ import { showError, useForm, useEffectAfterMount, useAsync, Progressing } from '
 import { toast } from 'react-toastify'
 import { List, CustomInput, ProtectedInput } from '../globalConfigurations/GlobalConfiguration'
 import Tippy from '@tippyjs/react';
-import { saveChartProviderConfig, updateChartProviderConfig, validateChartRepoConfiguration } from './chartRepo.service'
+import { saveChartProviderConfig, updateChartProviderConfig, validateChartRepoConfiguration, reSyncChartRepo } from './chartRepo.service';
 import { getChartRepoList } from '../../services/service'
 import { ReactComponent as Add } from '../../assets/icons/ic-add.svg';
 import { ReactComponent as Helm } from '../../assets/icons/ic-helmchart.svg';
 import { DOCUMENTATION } from '../../config';
 import { ValidateForm, VALIDATION_STATUS } from '../common/ValidateForm/ValidateForm';
+import "./chartRepo.scss";
 
 
 export default function ChartRepo() {
     const [loading, result, error, reload] = useAsync(getChartRepoList)
+    const [fetching, setFetching] = useState(false);
     if (loading && !result) return <Progressing pageLoader />
     if (error) {
         showError(error)
         if (!result) return null
     }
 
+    async function refetchCharts(e) {
+        if (fetching) {
+            return;
+        }
+        setFetching(true);
+        toast.success("Re-sync initiated. It may take upto 5 minutes for it to complete.")
+        await reSyncChartRepo().then((response) => {
+            setFetching(false);
+        }).catch((error) => {
+            showError(error);
+            setFetching(false);
+        })
+    }
+
     return (
         <section className="global-configuration__component">
             <h2 className="form__title">Chart Repository</h2>
             <p className="form__subtitle">Manage your organization’s chart repositories.
             <span><a rel="noreferrer noopener" target="_blank" className="learn-more__href" href={DOCUMENTATION.GLOBAL_CONFIG_CHART}> Learn more</a> </span></p>
-            {[{ id: null, default: true, url: "", name: "", active: true, authMode: "ANONYMOUS" }].concat(result && Array.isArray(result.result) ? result.result : []).sort((a, b) => a.name.localeCompare(b.name)).map(chart => <CollapsedList {...chart} key={chart.id || Math.random().toString(36).substr(2, 5)} reload={reload} />)}
+            <CollapsedList  id={null} default={true} url={""} name={""} active={true} authMode={"ANONYMOUS"}  key={Math.random().toString(36).substr(2, 5)} reload={reload} />
+            <div className="chartRepo_form__subtitle float-left bold">Repositories({result.result.length})</div>
```

Comment:
what if result is undefined - this code will fatal (checking at line no. 58)

Reply:
fixed